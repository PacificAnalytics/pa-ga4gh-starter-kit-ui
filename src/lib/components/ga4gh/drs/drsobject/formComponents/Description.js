import React from 'react';
import SimpleTextField from '../../../../common/SimpleTextField';

const Description = props => {
    return (
        <SimpleTextField
            id="description"
            label="Description"
            name="description"
            value={props.description}
            readOnly={props.readOnly}
            changeFunction={props.setDescription}
            helperText='Longer description of this DRS Object.'
        />
    )
}

export default Description;