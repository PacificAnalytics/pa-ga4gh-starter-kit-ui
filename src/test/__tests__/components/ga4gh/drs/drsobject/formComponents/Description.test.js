import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import {mockUpdateScalar} from '../../../../../../resources/MockFunctions';
import Description from '../../../../../../../lib/components/ga4gh/drs/drsobject/formComponents/Description';

afterEach(() => {
    mockUpdateScalar.mockClear();
})

test('SHOW <Description /> should handle populated description', () => {
    let {container} = render(
        <Description readOnly={true} description={'This is a description.'} 
        setDescription={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Description')).toBeInTheDocument();
    let descriptionField = screen.getByRole('textbox');
    expect(descriptionField).toHaveValue('This is a description.');

    // description field is read only and cannot be edited
    expect(descriptionField).toHaveAttribute('readonly');
    userEvent.type(descriptionField, 'test value');
    expect(mockUpdateScalar.mock.calls.length).toBe(0);
});

test('NEW and EDIT <Description /> should handle empty description', () => {
    let {container} = render(
        <Description readOnly={false} description={''} setDescription={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Description')).toBeInTheDocument();
    let descriptionField = screen.getByRole('textbox');
    expect(descriptionField).toHaveValue('');
    userEvent.type(descriptionField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});

test('NEW and EDIT <Description /> should handle populated description', () => {
    let {container} = render(
        <Description readOnly={false} description={'This is a description.'} 
        setDescription={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Description')).toBeInTheDocument();
    let descriptionField = screen.getByRole('textbox');
    expect(descriptionField).toHaveValue('This is a description.');
    userEvent.type(descriptionField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});