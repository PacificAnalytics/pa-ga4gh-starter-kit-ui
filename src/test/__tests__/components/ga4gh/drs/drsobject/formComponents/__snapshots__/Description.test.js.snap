// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <Description /> should handle empty description 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="description"
      id="description-label"
    >
      Description
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="description"
        name="description"
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;

exports[`NEW and EDIT <Description /> should handle populated description 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="description"
      id="description-label"
    >
      Description
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="description"
        name="description"
        type="text"
        value="This is a description."
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;

exports[`SHOW <Description /> should handle populated description 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="description"
      id="description-label"
    >
      Description
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="description-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="description"
        name="description"
        readonly=""
        type="text"
        value="This is a description."
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="description-helper-text"
    >
      Longer description of this DRS Object.
    </p>
  </div>
</div>
`;
